app/tray/Account/Account/Requests/SignPermitRequest/index.js

```diff
@@ -0,0 +1,223 @@
+import React, { useState, useEffect } from 'react'
+
+import DefaultSignature from '../SignTypedDataRequest/Default'
+import { formatDisplayInteger, isUnlimited } from '../../../../../../resources/utils/numbers'
+import svg from '../../../../../../resources/svg'
+import link from '../../../../../../resources/link'
+import { ClusterBox, Cluster, ClusterRow, ClusterValue } from '../../../../../../resources/Components/Cluster'
+import Countdown from '../../../../../../resources/Components/Countdown'
+import ChainHeader from '../../../../../../resources/Components/RequestChainHeader'
+import RequestItem from '../../../../../../resources/Components/RequestItem'
+import CustomAmountInput from '../../../../../../resources/Components/CustomAmountInput'
+
+const getPermit = (req) => {
+  const {
+    typedMessage: {
+      data: {
+        message: { deadline, spender, value, owner, nonce },
+        domain: { verifyingContract, chainId, name, version },
+        types
+      }
+    }
+  } = req
+
+  return {
+    deadline,
+    spender,
+    value,
+    owner,
+    verifyingContract,
+    chainId,
+    name,
+    nonce,
+    version,
+    types
+  }
+}
+
+const PermitSignature = ({ req, originName, step, chainData }) => {
+  const [tokenData, setTokenData] = useState(null)
+  useEffect(() => {
+    link.rpc('getErc20Data', verifyingContract, chainId, (err, tokenData) => {
+      //TODO: handle error state here
+      if (err) return console.error(err)
+      setTokenData(tokenData)
+    })
+  }, [])
+
+  const { deadline, spender, value, owner, verifyingContract, chainId, name, nonce, version, types } =
+    getPermit(req)
+  const { chainColor, chainName, icon } = chainData
+  const requestClass = getRequestClass(req)
+
+  const Overview = () => {
```

Comment:
you can't create new components inside another component's render function, they have to be declared outside

https://reactjs.org/docs/higher-order-components.html#dont-use-hocs-inside-the-render-method